### ISSUE TITLE
Migrate populate command to use ProgressBar

### ISSUE THREAD

**Issue by merk** [FriendsOfSymfony/FOSElasticaBundle#766]
Instead of using our own output, i'd prefer to see a ProgressBar if it is available.

**Comment by pmartelletti** [comment]
@merk something like this?

http://symfony.com/doc/2.3/components/console/helpers/progresshelper.html

If so, I will try to create a PR. :smile:

**Comment by merk** [comment]
The ProgressHelper is deprecated in favour of the ProgressBar, but the ProgressBar is not available in 2.3 or 2.4.

We still support symfony 2.3.

http://symfony.com/doc/current/components/console/helpers/progressbar.html

**Comment by pmartelletti** [comment]
@merk that's why I suggested using ProgressHelper instead of ProgressBar (that looks nicer). Or are you willing to find some external component?

**Comment by merk** [comment]
I dont want to implement a deprecated feature - we're going to have to evaluate 3.0 support in the next few months.

I'm not sure how hard it would be to use the ProgressBar when it is present or just default back to what we display at the moment if not..

**Comment by pmartelletti** [comment]
@merk in order to use the progress bar, I'll probably need to make the Provider method 'countObject' public. Is there a chance I can change that?

If not, progress bar can be used as a throbber instead of a progress bar.

http://symfony.com/doc/current/components/console/helpers/progressbar.html

**Comment by pmartelletti** [comment]
@merk I've made some tests for the ORM provider, and if we create a method like this:

``` php
/**
     * @return int|mixed
     */
    public function getTotalObjects()
    {
        return $this->countObjects($this->createQueryBuilder());
    }
```

we could implement the symfony progress bar, and it's working pretty well. Do you agree with this method (that could be implemented on the AbstractProvider)? Or do you have any ideas for a workaround?

**Comment by pmartelletti** [comment]
@merk created the PR. Let me know about it. :smile:

**Comment by royraider** [comment]
Hello !
I'm in Symfony 2.5.9 and I got this when trying to "populate":
php app/console fos:elastica:populate
> Resetting app                                    
  [LogicException]              
  You must start the progress bar before calling setCurrent().

I fixed this by disabling the ProgressBar. (line 128 of populateCommand)

**Comment by merk** [comment]
Please call the command with --verbose

**Comment by BourotBenjamin** [comment]
Here's what we got


     [LogicException]
     You must start the progress bar before calling setCurrent().

    Exception trace:
    () at C:\wamp\www\Youmiam\vendor\symfony\symfony\src\Symfony\Component\Console\Helper\ProgressBar.php:350
    Symfony\Component\Console\Helper\ProgressBar->setCurrent() at C:\wamp\www\Youmiam\vendor\symfony\symfony\src\Symfony\Component\Console\Helper\ProgressBar.php:337
    Symfony\Component\Console\Helper\ProgressBar->advance() at C:\wamp\www\Youmiam\vendor\friendsofsymfony\elastica-bundle\Command\PopulateCommand.php:171
    FOS\ElasticaBundle\Command\PopulateCommand->FOS\ElasticaBundle\Command\{closure}() at C:\wamp\www\Youmiam\vendor\friendsofsymfony\elastica-bundle\Doctrine\AbstractProvider.php:8
    FOS\ElasticaBundle\Doctrine\AbstractProvider->populate() at C:\wamp\www\Youmiam\vendor\friendsofsymfony\elastica-bundle\Command\PopulateCommand.php:114
    FOS\ElasticaBundle\Command\PopulateCommand->doPopulateType() at C:\wamp\www\Youmiam\vendor\friendsofsymfony\elastica-bundle\Command\PopulateCommand.php:198
    FOS\ElasticaBundle\Command\PopulateCommand->populateIndex() at C:\wamp\www\Youmiam\vendor\friendsofsymfony\elastica-bundle\Command\PopulateCommand.php:98
    FOS\ElasticaBundle\Command\PopulateCommand->execute() at C:\wamp\www\Youmiam\vendor\symfony\symfony\src\Symfony\Component\Console\Command\Command.php:253
    Symfony\Component\Console\Command\Command->run() at C:\wamp\www\Youmiam\vendor\symfony\symfony\src\Symfony\Component\Console\Application.php:896
    Symfony\Component\Console\Application->doRunCommand() at C:\wamp\www\Youmiam\vendor\symfony\symfony\src\Symfony\Component\Console\Application.php:193
    Symfony\Component\Console\Application->doRun() at C:\wamp\www\Youmiam\vendor\symfony\symfony\src\Symfony\Bundle\FrameworkBundle\Console\Application.php:96
    Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at C:\wamp\www\Youmiam\vendor\symfony\symfony\src\Symfony\Component\Console\Application.php:124
    Symfony\Component\Console\Application->run() at C:\wamp\www\Youmiam\app\console:24

**Comment by merk** [comment]
Please paste the output from `composer show -i` and `php -v`

**Comment by BourotBenjamin** [comment]
PHP 5.5.12 (cli) (built: Apr 30 2014 11:20:58)
    Copyright (c) 1997-2014 The PHP Group
    Zend Engine v2.5.0, Copyright (c) 1998-2014 Zend Technologies
        with Xdebug v2.2.5, Copyright (c) 2002-2014, by Derick Rethans


    a2lix/translation-form-bundle            dev-master a9d96ce Translate your doctrine objects easily with some helps
    behat/transliterator                     v1.0.1             String transliterator
    cocur/slugify                            v1.0               Converts a string into a slug.
    coupla/filepicker-bundle                 dev-master 740995e Symfony2 Framework integration with the Filpicker.io service
    craue/formflow-bundle                    dev-master 9e30495 Multi-step forms for your Symfony2 project.
    doctrine/annotations                     v1.2.3             Docblock Annotations Parser
    doctrine/cache                           v1.4.0             Caching library offering an object-oriented API for many cache backends
    doctrine/collections                     v1.2               Collections Abstraction library
    doctrine/common                          v2.4.2             Common Library for Doctrine projects
    doctrine/dbal                            v2.5.1             Database Abstraction Layer
    doctrine/doctrine-bundle                 dev-master 89196a1 Symfony DoctrineBundle
    doctrine/doctrine-cache-bundle           v1.0.1             Symfony2 Bundle for Doctrine Cache
    doctrine/inflector                       v1.0.1             Common String Manipulations with regard to casing and singular/plural rules.
    doctrine/instantiator                    1.0.4              A small, lightweight utility to instantiate objects in PHP without invoking their constructors
    doctrine/lexer                           v1.0.1             Base library for a lexer that can be used in Top-Down, Recursive Descent Parsers.
    doctrine/orm                             v2.4.7             Object-Relational-Mapper for PHP
    facebook/php-sdk-v4                      4.0.15             Facebook SDK for PHP
    friendsofsymfony/elastica-bundle         dev-master 7bc8849 Elasticsearch PHP integration for your Symfony2 project using Elastica
    friendsofsymfony/jsrouting-bundle        dev-master 1053dd4 A pretty nice way to expose your Symfony2 routing to client applications.
    friendsofsymfony/rest-bundle             1.4.2              This Bundle provides various tools to rapidly develop RESTful API's with Symfony2
    friendsofsymfony/user-bundle             dev-master b6a3b25 Symfony FOSUserBundle
    gedmo/doctrine-extensions                v2.3.11            Doctrine2 behavioral extensions
    genemu/form-bundle                       2.2.x-dev 2801f8a  Extra form types for your Symfony2 projects
    hwi/oauth-bundle                         dev-master a9f88f3 Support for authenticating users using both OAuth1.0a and OAuth2 in Symfony2.
    incenteev/composer-parameter-handler     dev-master 8f9608d Composer script handling your ignored parameter file
    jdorn/sql-formatter                      v1.2.17            a PHP SQL highlighting library
    jms/aop-bundle                           1.0.1              Adds AOP capabilities to Symfony2
    jms/cg                                   1.0.0              Toolset for generating PHP code
    jms/di-extra-bundle                      1.4.0              Allows to configure dependency injection using annotations
    jms/i18n-routing-bundle                  dev-master 6f69173 This bundle allows you to create i18n routes.
    jms/metadata                             1.5.1              Class/method/property metadata management in PHP
    jms/parser-lib                           1.0.0              A library for easily creating recursive-descent parsers.
    jms/security-extra-bundle                dev-master 59b777e Enhances the Symfony2 Security Component by adding several new features
    jms/serializer                           0.16.0             Library for (de-)serializing data of any complexity; supports XML, JSON, and YAML.
    jms/serializer-bundle                    0.12.0             Allows you to easily serialize, and deserialize data of any complexity
    jms/translation-bundle                   1.1.0              Puts the Symfony2 Translation Component on steroids
    kertz/twitteroauth                       dev-master 97b677c
    knplabs/doctrine-behaviors               v1.0.2             Doctrine2 behavior traits
    knplabs/knp-components                   1.3.1              Knplabs component library
    knplabs/knp-menu                         dev-master d6f3cc4 An object oriented menu library
    knplabs/knp-menu-bundle                  dev-master dc2c5b9 This bundle provides an integration of the KnpMenu library
    knplabs/knp-paginator-bundle             dev-master 5990419 Paginator bundle for Symfony2 to automate pagination and simplify sorting and other features
    kriswallsmith/assetic                    1.2.x-dev b20efe3  Asset Management for PHP
    kriswallsmith/buzz                       v0.13              Lightweight HTTP client
    michelf/php-markdown                     1.4.1              PHP Markdown
    monolog/monolog                          1.12.0             Sends your logs to files, sockets, inboxes, databases and various web services
    mopa/bootstrap-bundle                    2.3.x-dev a71200d  Easy integration of twitters bootstrap into symfony2
    mopa/composer-bridge                     v1.3.1             Symfony2 Composer Bridge
    nelmio/api-doc-bundle                    2.7.0              Generates documentation for your REST API from annotations
    nikic/php-parser                         v0.9.1             A PHP parser written in PHP
    oldsound/rabbitmq-bundle                 v1.2.1             Integrates php-amqplib with Symfony2 and RabbitMq
    phpcollection/phpcollection              0.4.0              General-Purpose Collection Library for PHP
    phpdocumentor/reflection-docblock        2.0.3
    phpoption/phpoption                      1.4.0              Option Type for PHP
    phpunit/php-code-coverage                2.0.15             Library that provides collection, processing, and rendering functionality for PHP code coverage information.
    phpunit/php-file-iterator                1.3.4              FilterIterator implementation that filters files based on a list of suffixes.
    phpunit/php-text-template                1.2.0              Simple template engine.
    phpunit/php-timer                        1.0.5              Utility class for timing
    phpunit/php-token-stream                 1.4.0              Wrapper around PHP's tokenizer extension.
    phpunit/phpunit                          4.2.6              The PHP Unit Testing framework.
    phpunit/phpunit-mock-objects             2.3.0              Mock Object library for PHPUnit
    psr/log                                  1.0.0              Common interface for logging libraries
    raulfraile/ladybug                       v0.7               Simple and Extensible PHP Dumper
    raulfraile/ladybug-bundle                v0.7               Symfony2 bundle for Ladybug library (Simple and Extensible PHP Dumper)
    richsage/rms-push-notifications-bundle   0.1.2              Push notifications/messages for mobile devices
    ruflin/elastica                          v1.3.4.0           Elasticsearch Client
    sebastian/comparator                     1.1.1              Provides the functionality to compare PHP values for equality
    sebastian/diff                           1.2.0              Diff implementation
    sebastian/environment                    1.2.1              Provides functionality to handle HHVM/PHP environments
    sebastian/exporter                       1.2.0              Provides the functionality to export PHP variables for visualization
    sebastian/recursion-context              1.0.0              Provides functionality to recursively process PHP variables
    sebastian/version                        1.0.4              Library that helps with managing the version number of Git-hosted PHP projects
    segmentio/analytics-php                  1.0.3              Segmentio Analytics PHP Library
    sensio/distribution-bundle               v2.3.9             The base bundle for the Symfony Distributions
    sensio/framework-extra-bundle            dev-master b2c9d85 This bundle provides a way to configure your controllers with annotations
    sensio/generator-bundle                  v2.5.1             This bundle generates code for you
    sonata-project/admin-bundle              2.3.x-dev 95a6bc5  Symfony SonataAdminBundle
    sonata-project/block-bundle              2.2.x-dev f4d9658  Symfony SonataBlockBundle
    sonata-project/cache                     1.0.4              Cache library
    sonata-project/cache-bundle              dev-master 4b30f1b This bundle provides caching services
    sonata-project/core-bundle               dev-master ab941c3 Symfony SonataCoreBundle
    sonata-project/doctrine-orm-admin-bundle 2.3.x-dev 54d4b79  Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
    sonata-project/exporter                  1.3.4              Lightweight Exporter library
    stof/doctrine-extensions-bundle          v1.1.0             Integration of the gedmo/doctrine-extensions with Symfony2
    swiftmailer/swiftmailer                  v5.3.1             Swiftmailer, free feature-rich PHP mailer
    symfony/assetic-bundle                   v2.4.0             Integrates Assetic into Symfony2
    symfony/css-selector                     2.6.x-dev 3f80ecc  Symfony CssSelector Component
    symfony/monolog-bundle                   v2.6.1             Symfony MonologBundle
    symfony/stopwatch                        dev-master dbbf321 Symfony Stopwatch Component
    symfony/swiftmailer-bundle               v2.3.8             Symfony SwiftmailerBundle
    symfony/symfony                          v2.5.9             The Symfony PHP framework
    twbs/bootstrap                           v2.3.2             Sleek, intuitive, and powerful front-end framework for faster and easier web development.
    twig/extensions                          v1.2.0             Common additional features for Twig that do not directly belong in core
    twig/twig                                v1.18.0            Twig, the flexible, fast, and secure template language for PHP
    videlalvaro/php-amqplib                  v2.1.0             This library is a pure PHP implementation of the AMQP protocol. It's been tested against RabbitMQ.
    willdurand/js-translation-bundle         v2.1.4             A pretty nice way to expose your translation messages to your JavaScript.
    willdurand/jsonp-callback-validator      v1.1.0             JSONP callback validator.
    willdurand/negotiation                   1.3.4              Content Negotiation tools for PHP provided as a standalone library.

**Comment by merk** [comment]
The latest commit should fix your problem.

https://github.com/FriendsOfSymfony/FOSElasticaBundle/commit/72589f8341d9dc3ad78f07a45fb2b8a158a3c350

**Comment by BourotBenjamin** [comment]
There still a problem : 

    C:\wamp\www\MyApp>php app/console fos:elastica:populate
      100/1997 [=>--------------------------]   5%
    Populating myapp/product
    Fatal error: Call to undefined method Symfony\Component\Console\Helper\ProgressBar::getProgress() in C:\wamp\www\MyApp\vendor\friendsofsymfony\elastica-bundle\Command\PopulateCom
    mand.php on line 173

    Call Stack:
        0.0010     235640   1. {main}() C:\wamp\www\MyApp\app\console:0
        0.0380    3578544   2. Symfony\Component\Console\Application->run() C:\wamp\www\MyApp\app\console:24
        0.0440    3810992   3. Symfony\Bundle\FrameworkBundle\Console\Application->doRun() C:\wamp\www\MyApp\vendor\symfony\symfony\src\Symfony\Component\Console\Application.php:124
        1.6241   19463032   4. Symfony\Component\Console\Application->doRun() C:\wamp\www\MyApp\vendor\symfony\symfony\src\Symfony\Bundle\FrameworkBundle\Console\Application.php:96
        1.6251   19463872   5. Symfony\Component\Console\Application->doRunCommand() C:\wamp\www\MyApp\vendor\symfony\symfony\src\Symfony\Component\Console\Application.php:193
        1.6291   19546544   6. Symfony\Component\Console\Command\Command->run() C:\wamp\www\MyApp\vendor\symfony\symfony\src\Symfony\Component\Console\Application.php:896
        1.6451   20292024   7. FOS\ElasticaBundle\Command\PopulateCommand->execute() C:\wamp\www\MyApp\vendor\symfony\symfony\src\Symfony\Component\Console\Command\Command.php:253
        1.6451   20295384   8. FOS\ElasticaBundle\Command\PopulateCommand->populateIndex() C:\wamp\www\MyApp\vendor\friendsofsymfony\elastica-bundle\Command\PopulateCommand.php:98
      301.7093   21326928   9. FOS\ElasticaBundle\Command\PopulateCommand->doPopulateType() C:\wamp\www\MyApp\vendor\friendsofsymfony\elastica-bundle\Command\PopulateCommand.php:198
      301.7123   21461544  10. FOS\ElasticaBundle\Doctrine\AbstractProvider->populate() C:\wamp\www\MyApp\vendor\friendsofsymfony\elastica-bundle\Command\PopulateCommand.php:114
      305.8405   77930704  11. FOS\ElasticaBundle\Command\PopulateCommand->FOS\ElasticaBundle\Command\{closure}() C:\wamp\www\MyApp\vendor\friendsofsymfony\elastica-bundle\Doctrine\AbstractProvider.php:83

**Comment by merk** [comment]
That one should be fixed now too.

**Comment by merk** [comment]
Implemented in 3.1-dev
